graphjet-core/src/test/java/com/twitter/graphjet/bipartite/edgepool/RegularDegreeEdgePoolTest.java

```diff
@@ -195,4 +201,67 @@ public void testRandomConcurrentReadWrites() {
     testRandomConcurrentReadWriteThreads(
         regularDegreeEdgePool, 3, 10 * maxNumNodes, maxDegree, 0.1, random);
   }
+
+  @Test
+  public void testGetNodeEdgeMetadataReturningNegative() {
+      NullStatsReceiver nullStatsReceiver = new NullStatsReceiver();
+      RegularDegreeEdgePool regularDegreeEdgePool = new RegularDegreeEdgePool(4, 4, nullStatsReceiver);
+      regularDegreeEdgePool.addEdge(2048, 587, (-424L));
+
+      assertEquals((-1L), regularDegreeEdgePool.getNodeEdgeMetadata(2048, 426));
+ }
+
+  @Test
+  public void testAddEdgeTakingThreeArgumentsThrowsIllegalArgumentException() {
+      NullStatsReceiver nullStatsReceiver = new NullStatsReceiver();
+      RegularDegreeEdgePool regularDegreeEdgePool = new RegularDegreeEdgePool(508, 2, nullStatsReceiver);
+      ShardedBigIntArray shardedBigIntArray = new ShardedBigIntArray(3157, 2, (-1035), nullStatsReceiver);
+      ShardedBigLongArray shardedBigLongArray = new ShardedBigLongArray(1431655765, 2, 2, nullStatsReceiver);
+      IntToIntPairArrayIndexBasedMap intToIntPairArrayIndexBasedMap = new IntToIntPairArrayIndexBasedMap(3157, 16, nullStatsReceiver);
+      RegularDegreeEdgePool.ReaderAccessibleInfo regularDegreeEdgePool_ReaderAccessibleInfo = new RegularDegreeEdgePool.ReaderAccessibleInfo(shardedBigIntArray, shardedBigLongArray, intToIntPairArrayIndexBasedMap);
```

Comment:
nitpick: in GraphJet code base, we prefer to have each line of code shorter than 100 characters. for example, in this line, it is better to break it into two or three different lines.